api/controllers/game/rematch.js

```diff
@@ -21,17 +26,16 @@ module.exports = async function (req, res) {
 
     const { p0: newP1Id, p1: newP0Id } = updatedGame;
 
-    Game.publish([game.id], {
```

Comment:
The endpoint should only emit a single socket event. Previously it would emit one to say that a user had opted for/against a rematch, then another one when the game started (if both users said yes). But that creates a race condition between the two socket events triggered by the same endpoint. 

This way there is just one socket event that fires for any given code path through this endpoint, and it's either `'rematch'` which means we're just updating a rematch card, or it's `'newGameForRematch'`, which means the game started. When it's the new game, the client handles the animation for flipping the card before processing the new game and navigating to the new game url